### PR TITLE
refactor(mdxprovider): add MDXProvider pattern

### PR DESCRIPTION
**Pull request by maiieul**
# What is it?

- [x] Feature / enhancement
- [ ] Bug
- [x] Docs / tests

# Description

This PR adds the MDXProvider pattern from #494 but doesn't implement the import.meta.glob pattern (I will do another PR for that).

- Added `<MDXProvider>` to root layout with `components` object from `mdx-components.tsx`, which is possible thanks to `recma-provide-components.tsx` script (implemented through vite.config.ts) 
- added `rehypePrettyCode` plugin (also used by qwik's docs) with two small rehype scripts for Copying the Code without having to use a component in the markdown.
- Rewrote .mdx routes to remove the now unnecessary imports.

I Manually checked that everything works on `pnpm dev`, `pnpm preview` and `pnpm preview.cloudflare`.

# Checklist:

- [x] My code follows the [developer guidelines of this project](https://github.com/qwikifiers/qwik-ui/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] Added new tests to cover the fix / functionality

### CODE REVIEW

File: apps/website/src/routes/docs/headless/(components)/navigation-bar/index.tsx [hunk 1]
```diff
@@ -9,12 +9,12 @@ export default component$(() => {
...
 
-      <div class="flex flex-col gap-8 mt-4">
+      <div class="mt-4 flex flex-col gap-8">
         <h2>Basic Example</h2>
 
-        <NavigationBar class="items-center navigation-bar flex w-full bg-base-100 rounded-lg">
+        <NavigationBar class="navigation-bar bg-base-100 flex w-full items-center rounded-lg">
           <div class="navigation-bar__start flex">
-            <a class="btn btn-ghost normal-case text-xl ">tailwindUI</a>
+            <a class="btn btn-ghost text-xl normal-case ">tailwindUI</a>
```
Comment (thejackshelton) [inline apps/website/src/routes/docs/headless/(components)/navigation-bar/index.tsx:17]:
```suggestion
            <a href="" class="btn btn-ghost text-xl normal-case ">tailwindUI</a>
```

let's add an href here so we don't get a warning from qwik

File: apps/website/src/global.css [hunk 2]
```diff
@@ -154,12 +154,6 @@
       var(--qwikui-slate-900)
     );
   }
```
Comment (thejackshelton) [inline apps/website/src/global.css:156]: Add this back, otherwise we get horizontal overflow on the code examples.

File: package.json [hunk 3]
```diff
@@ -101,17 +102,21 @@
     "prettier": "^2.8.8",
     "prettier-plugin-tailwindcss": "0.4.1",
     "pretty-quick": "^3.1.3",
+    "rehype-pretty-code": "^0.10.2",
     "qwik-nx": "^1.0.8",
     "react": "18.2.0",
     "react-dom": "18.2.0",
     "sass": "^1.66.1",
+    "shiki": "0.14.5",
```
Comment (thejackshelton) [inline package.json:110]: I don't think we are using shiki anymore, let's remove this.

File: apps/website/src/global.css (hunk 2 above)
Comment (thejackshelton) [reply apps/website/src/global.css:156]: image

**Comment by thejackshelton** [comment]
@maiieul is killin it! If we could make those couple of changes.

Also since we're making a decent bit of changes to the vite config of the website let's have a second reviewer.

Looks good to me!

**Review by gioboa — APPROVED** [review]
It looks great 👍

**Pull request merged**